cpp/src/arrow/testing/random.h

```diff
@@ -49,7 +50,9 @@ class ARROW_TESTING_EXPORT RandomArrayGenerator {
   /// \param[in] null_probability the probability of a bit being zero
   ///
   /// \return a generated Buffer
-  std::shared_ptr<Buffer> NullBitmap(int64_t size, double null_probability = 0);
+  std::shared_ptr<Buffer> NullBitmap(int64_t size, double null_probability = 0,
+                                     int64_t alignment = MemoryPool::kDefaultAlignment,
+                                     MemoryPool* memory_pool = NULLPTR);
```

Comment:
I did want to use these methods to generate test data for the spilling classes in the spilling PR which assume aligned buffers. It would be possible to work around of course, but having this in place makes those tests much more convenient.